### PR TITLE
HARP-17671: Fixes Frame Complete

### PR DESCRIPTION
**Pull request by FraukeF**
…after finished TextPlacement

Signed-off-by: Frauke Fritz <frauke.fritz@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Review by ThFischer — APPROVED** [review]

**Comment by FraukeF** [comment]
retest please

**Comment by FraukeF** [comment]
Rebase

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts [hunk 1]
```diff
@@ -908,7 +921,7 @@ export class TextElementsRenderer {
 
             if (elevationProvider !== undefined && !textElement.elevated) {
                 if (!elevationMap) {
-                    this.m_viewUpdateCallback(); // Update view until elevation is loaded.
+                    this.m_isUpdatePending = true;
```
Comment (atomicsulfate) [inline @here/harp-mapview/lib/text/TextElementsRenderer.ts:924]: Before we were actively requesting a MapView update to ensure there will be a follow-up frame. Now we're only passively marking the TextElementsRenderer as pending for update. Who request the next animation frame now?
Sample applies to all places where you changed the view update callback.

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts [hunk 2]
```diff
@@ -1739,11 +1750,6 @@ export class TextElementsRenderer {
             if (iconInvisible) {
                 iconRenderState.reset();
             }
-        } else if (renderIcon && poiInfo!.isValid !== false) {
-            // Ensure that text elements still loading icons get a chance to be rendered if
-            // there are no text element updates in the next frames.
-            this.m_forceNewLabelsPass = true;
-            this.m_viewUpdateCallback();
```
Comment (atomicsulfate) [inline @here/harp-mapview/lib/text/TextElementsRenderer.ts:1746]: what happens with icon images that are still loading?

**Review by atomicsulfate — CHANGES_REQUESTED** [review]

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts (hunk 2 above)
Comment (FraukeF) [reply @here/harp-mapview/lib/text/TextElementsRenderer.ts:1746]: they were already covered by the loading state, which was not evaluated correctly due to this unfortunate flag in  MapView. Test are all passing and as far as I see everything still works. I ll keep an eye on it

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts (hunk 1 above)
Comment (FraukeF) [reply @here/harp-mapview/lib/text/TextElementsRenderer.ts:924]: Basically Mapview now checks for needed updates, as it already did for loading of the fonts and also does for other depending instances like the VisibleTiles, ExtrusionHandler, AnimationFramework, etc too.
Actually I think we should do this throughout the code and use MapView.update only as external API. 
Currently MapView is driving the RenderLoop and then we have some inconsistencies on which components are requesting updates, which are requested for updates.. and TextElementsRenderer actually had both :/

File: @here/harp-mapview/lib/MapView.ts [hunk 3]
```diff
@@ -3453,22 +3458,6 @@ export class MapView extends EventDispatcher {
...
-            !this.m_themeManager.isUpdating() &&
-            this.m_poiTableManager.finishedLoading &&
-            this.m_visibleTiles.allVisibleTilesLoaded &&
-            this.m_connectedDataSources.size + this.m_failedDataSources.size ===
-                this.m_tileDataSources.length &&
-            !this.m_textElementsRenderer.loading
-        ) {
-            this.m_initialTextPlacementDone = true;
-        }
-
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:3471]: very very welcome, its very satisfiing to delete it :)

File: @here/harp-mapview/lib/MapView.ts [hunk 4]
```diff
@@ -2403,12 +2404,16 @@ export class MapView extends EventDispatcher {
     /**
      * Returns `true` if the current frame will immediately be followed by another frame.
      */
-    get isDynamicFrame(): boolean {
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:2406]: What about I keep the old name and deprecate it, also the new isUpdatePending probably should not be API

[the remark this answers is not in the document]
Comment (atomicsulfate) [reply @here/harp-mapview/lib/MapView.ts:2406]: sounds good to me. So let's deprecate isDynamicFrame. isUpdatePending() can even be private.

**Review by atomicsulfate — APPROVED** [review]

**Pull request merged**